daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala

```diff
@@ -1617,6 +1637,26 @@ object SBuiltinTest {
           val from1 : AnyException -> Text = \(e:AnyException) -> case from_any_exception @Mod:Ex1 e of None -> "NONE" | Some x -> Mod:Ex1 { message} x;
           val from2 : AnyException -> Text = \(e:AnyException) -> case from_any_exception @Mod:Ex2 e of None -> "NONE" | Some x -> Mod:Ex2 { message} x;
           val from3 : AnyException -> Text = \(e:AnyException) -> case from_any_exception @Mod:Ex3 e of None -> "NONE" | Some x -> Mod:Ex3 { message} x;
+
+          interface (this : Iface) = {
+              precondition True;
+          };
+
+          record @serializable Iou = { i: Party, u: Party, name: Text };
+          template (this: Iou) = {
+            precondition True;
+            signatories Cons @Party [Mod:Iou {i} this] (Nil @Party);
+            observers Cons @Party [Mod:Iou {u} this] (Nil @Party);
+            agreement "Agreement";
+            implements Mod:Iface {};
+          };
+
+          val mkParty : Text -> Party = \(t:Text) -> case TEXT_TO_PARTY t of None -> ERROR @Party "none" | Some x -> x;
+          val alice : Party = Mod:mkParty "alice";
+          val bob : Party = Mod:mkParty "bob";
+
+          val aliceOwesBob : Mod:Iou = Mod:Iou { i = Mod:alice, u = Mod:bob, name = "alice owes bob" };
+          val aliceOwesBobIface : Mod:Iface = to_interface @Mod:Iface @Mod:Iou Mod:aliceOwesBob;
```

Comment:
The file start to be really big. it may be nice to start a new one. At least I will keep this file for the LF builtins, and start another file for those internal builtin that depends of the packages (update, exception...).
No need to tackle that in this PR.